### PR TITLE
Add lisk-db package - Closes #5241 and #5242

### LINKED ISSUE #5241: Create lisk-db package in elements
### Description

With the introduction of an `embedded` database system, a new elements package named `lisk-db` should be created.

### Acceptance Criteria

- Create `lisk-db` package from template
- Implement `KVStore` class exposing public interfaces
- Add unit tests

#### Interface

```
class KVStore {
	public constructor(filePath: string)

	public async get(key: string): Promise<any>

	public async put(key: string, val: Buffer): Promise<void>

	public async del(key: string): Promise<void>

	public async exists(key: string): Promise<boolean>

	public batch(): <T> # T -> {put, write, del}

	public createReadStream(options?: Options): ReadStream

	public async clear(): Promise<void>

        public async close(): Promise<void>
}
```

Related issues: #5238

### LINKED ISSUE #5242: Create utility to get a range pointer
### Description

In order to get a range of data for particular type of key we need a utility function for an `embedded` database to get data for a given range.

- Add function to calculate next binary from the prefix
- Add function to calculate before binary from the prefix

### Motivation

With the introduction of an `embedded` database system we need a mechanism to get data in a range.

### Acceptance Criteria

- Create util function to get range pointer
- Add unit tests to get range data

#### Example:

```js
block:height:0000001
block:height:0000002
block:height:0000003
block:height:0000004
```

```js
db.getBlockBetweenHeight({
  gte: from_height,
  lte: range_pointer
})
```

Related issues: #5238

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #5241 and resolves #5242 

### How was it solved?

- Add `lisk-db` package similar to other packages
- Add basic implementation of the interface
- Add test covering the implementation

### How was it tested?

- All the usages are covered by test

### CODE REVIEW

File: elements/lisk-db/src/error.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
```
Comment (pablitovicente) [inline elements/lisk-db/src/error.ts:14]: If in the future we need different errors this file will export new classes?

File: elements/lisk-db/src/kv_store.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import * as path from 'path';
+import * as fs from 'fs';
+import { debug } from 'debug';
+import encodingDown from 'encoding-down';
```
Comment (pablitovicente) [inline elements/lisk-db/src/kv_store.ts:17]: Do we need this if lisk-codec is going to be implemented?

Comment (shuse2) [reply elements/lisk-db/src/kv_store.ts:17]: when we integrate codec, this needs to be removed along with changing the interface to Buffer

File: elements/lisk-db/src/error.ts (hunk 1 above)
Comment (shuse2) [reply elements/lisk-db/src/error.ts:14]: If we need them, yes

File: elements/lisk-db/src/kv_store.ts (hunk 2 above)
Comment (pablitovicente) [reply elements/lisk-db/src/kv_store.ts:17]: OK maybe leave a comment?

File: elements/lisk-db/src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+		buf.writeBigInt64BE(num);
+	} else {
+		buf = Buffer.alloc(4);
+		buf.writeInt32BE(num);
+	}
+	return buf.toString('binary');
+};
+
+export const getStartPrefix = (prefix: string): string => `${prefix}\x00`;
+export const getEndPrefix = (prefix: string): string => `${prefix}\xFF`;
```
Comment (ishantiw) [inline elements/lisk-db/src/utils.ts:28]: maybe change it to `getSuffix`?

File: elements/lisk-db/test/utils.spec.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { formatInt, getStartPrefix, getEndPrefix } from '../src/utils';
+
+describe('utils', () => {
+	describe('formatInt', () => {
+		describe('when bigint is provided', () => {
+			it('should return string which can be sorted lexigraphically', () => {
```
Comment (ishantiw) [inline elements/lisk-db/test/utils.spec.ts:20]: `lexicographically`

File: elements/lisk-db/test/utils.spec.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+					'000000000000000b',
+				);
+				expect(Buffer.from(stringArray[2], 'binary').toString('hex')).toEqual(
+					'0000000000000064',
+				);
+			});
+		});
+
+		describe('when number is provided', () => {
+			it('should return string which can be sorted lexigraphically', () => {
```
Comment (ishantiw) [inline elements/lisk-db/test/utils.spec.ts:40]: `lexicographically`

File: elements/lisk-db/src/utils.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+		buf = Buffer.alloc(8);
+		buf.writeBigInt64BE(num);
+	} else {
+		buf = Buffer.alloc(4);
+		buf.writeInt32BE(num);
+	}
+	return buf.toString('binary');
+};
+
+export const getStartPrefix = (prefix: string): string => `${prefix}\x00`;
```
Comment (ishantiw) [inline elements/lisk-db/src/utils.ts:27]: also, this can be changed to `getPrefix` if we change the other one to `getSuffix`

Comment (shuse2) [reply elements/lisk-db/src/utils.ts:27]: But it's not getting prefix or suffix.
It more like getting first key possible of the prefix, and getting last key possible of the prefix

Comment (ishantiw) [reply elements/lisk-db/src/utils.ts:27]: the `getLastPrefix` sounds more correct

Comment (ishantiw) [reply elements/lisk-db/src/utils.ts:27]: `getFirstPrefix` and `getLastPrefix `

**Pull request merged**